src/viam/ur/module/ur_arm.cpp

```diff
@@ -796,23 +811,50 @@ void URArm::keep_alive_() {
 void URArm::move_(std::shared_lock<std::shared_mutex> config_rlock, std::list<Eigen::VectorXd> waypoints, const std::string& unix_time) {
     auto our_config_rlock = std::move(config_rlock);
 
+    // get current joint position and add that as starting pose to waypoints
+    VIAM_SDK_LOG(info) << "move: get_joint_positions start " << unix_time;
+    auto curr_joint_pos_deg = get_joint_positions_(our_config_rlock);
+    VIAM_SDK_LOG(info) << "move: get_joint_positions end " << unix_time;
+    auto curr_joint_pos_rad = Eigen::VectorXd(curr_joint_pos_deg.size());
+    std::transform(std::make_move_iterator(curr_joint_pos_deg.begin()),
+                   std::make_move_iterator(curr_joint_pos_deg.end()),
+                   curr_joint_pos_rad.begin(),
+                   degrees_to_radians<double>);
+
+    if (reject_move_request_threshold_deg) {
+        auto delta_pos = (waypoints.front() - curr_joint_pos_rad);
+
+        if (radians_to_degrees(delta_pos.lpNorm<Eigen::Infinity>()) > *reject_move_request_threshold_deg) {
+            std::stringstream err_string;
+
+            err_string << "rejecting move request : difference between starting trajectory position [(";
+            std::transform(std::make_move_iterator(waypoints.front().cbegin()),
+                           std::make_move_iterator(waypoints.front().cend()),
+                           std::experimental::make_ostream_joiner(err_string, ", "),
+                           radians_to_degrees<const double>);
+            err_string << ")] and joint position [(";
+            std::transform(std::make_move_iterator(curr_joint_pos_rad.cbegin()),
+                           std::make_move_iterator(curr_joint_pos_rad.cend()),
+                           std::experimental::make_ostream_joiner(err_string, ", "),
+                           radians_to_degrees<const double>);
+            err_string << ")] is above threshold " << radians_to_degrees(delta_pos.lpNorm<Eigen::Infinity>()) << " > "
+                       << *reject_move_request_threshold_deg;
+            VIAM_SDK_LOG(error) << err_string.str();
+            throw std::runtime_error(err_string.str());
+        }
+    }
+
     // Capture the current movement epoch, so we can later detect if another caller
     // slipped in while we were planning.
     auto current_move_epoch = current_state_->move_epoch.load(std::memory_order_acquire);
 
     VIAM_SDK_LOG(info) << "move: start unix_time_ms " << unix_time << " waypoints size " << waypoints.size();
     const auto log_move_end = make_scope_guard([&] { VIAM_SDK_LOG(info) << "move: end unix_time " << unix_time; });
 
-    // get current joint position and add that as starting pose to waypoints
```

Comment:
Was there a reason you effectively moved this code up above the `current_move_epoch` acquisition and the `log_move_end` guard to add the check? I think it would be more natural to insert it down here (it'd make the diff a little clearer too).

Reply:
nope